pytorch-notebook/environment.yml

```diff
@@ -0,0 +1,12 @@
+# Note: this environment will be merged with the pangeo-notebook/environment.yml
+# by conda-lock. Only put things here that are not already in pangeo-notebook
+name: pangeo
+channels:
+ - conda-forge
+dependencies:
+ - cudatoolkit=11
```

Comment:
in the ml-notebook image we pin cudatoolkit=10, https://github.com/pangeo-data/pangeo-docker-images/blob/b5c38d09ca44e66dfda529fbfeb15ba4789a3805/ml-notebook/environment.yml#L7

I'm guessing if both these images are offered on pangeo jupyterhubs we want cuda versions to be the same, @yuvipanda any recommendations for 10 vs 11 when it comes to node configuration? 

Fine leaving it as 11 here and bumping the other image in a separate PR.

Reply:
Yes, it comes down to what CUDA drivers the server is using (find out using `nvidia-smi`). I prefer `cudatoolkit=11` because it has foward and backward compatibility within a minor version (see https://docs.nvidia.com/deploy/cuda-compatibility/#forward-compatibility-title), but that's of no use if the cuda drivers on the server don't support it.

Reply:
I don't actually know enough to have an opinion here, but happy to stick to whatever upstream recommends and as long as we are consistent across the images :D If we need to upgrade the driver version on the clusters when we bump this, can do that.

Reply:
Ok. Just to confirm, I'm assuming that the clusters don't pull the latest pangeo-docker image? I.e. that they're pinned to a specific pangeo-docker-image version? Probably doesn't matter for this `pytorch-notebook` image since nobody will be using it yet, but I'd recommend updating the CUDA driver version before bumping the `cudatoolkit` version on `ml/tensorflow-notebook`, just speaking from experience working on an on-prem HPC :slightly_smiling_face:

Reply:
> clusters don't pull the latest pangeo-docker image? 

Actually they do currently pull the latest :) https://github.com/2i2c-org/infrastructure/search?q=ml-notebook 
But I say we go ahead and merge this, and can use explicit pins on the hub if need be going forward

Reply:
Wow, you all like to live dangerously :laughing: So looking at https://github.com/2i2c-org/infrastructure/pull/1244, it seems like the GPUs are K80s, and according to https://forums.developer.nvidia.com/t/nvidia-tesla-k80-cuda-version-support/67676/6, CUDA 11 to work but support is deprecated (i.e. there might be lots of warnings). So let's test it out then!

Reply:
We are not currently using ml-notebook in producing in 2i2c. All production clusters use pinned tags.

Reply:
One other point on cudatoolkit 10 vs 11 is that 11 brings along a few more gigabytes. I don't mind big images, but this one is currently approaching 10GB total uncompressed...

`mamba create -n test --dry-run pytorch-gpu cudatoolkit=10` -> Total download: 1 GB
`mamba create -n test --dry-run pytorch-gpu cudatoolkit=11` ->  Total download: 3 GB

Reply:
Yes cudatoolkit is huge unfortunately. There are ways to trim down the docker image size as you mentioned in https://github.com/pangeo-data/pangeo-docker-images/issues/188#issuecomment-788447708, but it's tough getting around a big binary... I think CUDA 11 will need to be used eventually though, so will need to figure out a solution :upside_down_face: